set.go

```diff
@@ -133,6 +136,25 @@ func (s *lockingSet[T]) ToSlice() []T {
 	return slice
 }
 
+// Filtered returns a slice of items that match the predicate.
+func (s *lockingSet[T]) Filtered(predicate func(T) bool) []T {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	l := len(s.m)
+	if l == 0 {
+		return nil
+	}
+
+	var slice []T
+	for k := range s.m {
+		if predicate(k) {
+			slice = append(slice, k)
+		}
+	}
```

Comment:
So, I didn't put much additional thought into this, but it did occur to me that we could always create a slice with capacity of `len(s.m)` which would decrease the total number of reallocs that would occur on append, but we'd almost always over-allocate if there were actual filtered events. Given that this number is small most of the time, I don't really think it's a huge problem, but worth a passing thought if you have one :)